onnxruntime/core/providers/cpu/signal/dft.h

```diff
@@ -19,25 +20,14 @@ class DFT final : public OpKernel {
   Status Compute(OpKernelContext* ctx) const override;
```

Comment:
It seems the default value for axis is not the same for domain '' and domain com.microsoft. Should it be `axis_ = info.GetAttrOrDefault<int64_t>("axis", 0 or 1 based on the domain);`?

Reply:
In onnx-script, I added unit tests to check that every configuration (1D, axis=0, 2D, axis=-1, 0 or 1, 3D, axis=-1, 0, 1, 2) is working to make sure the code was not trying to access a number out of range.

Reply:
There is a utility function that must be used. HandleNegativeAxis()

Reply:
IIRC this was handled in the contrib op in the implementation .cc with:
https://github.com/microsoft/onnxruntime/blob/80d8c4c7ff4075495087ecde08892113940a5d24/onnxruntime/contrib_ops/cpu/signal/dft.cc#L367

auto rank = static_cast<int64_t>(X_shape.GetDims().size());
  if (!(-rank <= axis && axis < rank)) {
    ORT_RETURN_IF(!(-rank <= axis && axis < rank),
                  "axis attribute value ",
                  axis,
                  " is invalid for a tensor of rank ",
                  rank);
  }
  axis = (axis >= 0 ? axis : axis + rank);

Good that there is a helper here for it.